chalk-macros/src/index.rs

```diff
@@ -22,6 +22,11 @@ macro_rules! index_struct {
             $v fn increment(&mut self) {
                 self.value += 1;
             }
+
+            // TODO: Once the Step trait is stabilized (https://github.com/rust-lang/rust/issues/42168), instead implement it and use the Iterator implementation of Range
+            pub fn iterate_range(range: ::std::ops::Range<Self>) -> impl Iterator<Item = $n> {
```

Comment:
Nit: Can we make this a method with an extension trait? would read nicer.

Reply:
Hm, I don't think I'd be able to use impl trait in that case?

Reply:
Yeah, not on stable, at least.